scene/2d/camera_2d.cpp

```diff
@@ -453,15 +462,25 @@ void Camera2D::make_current() {
 
 void Camera2D::clear_current() {
 	ERR_FAIL_COND(!is_current());
-	if (viewport && !(custom_viewport && !ObjectDB::get_instance(custom_viewport_id)) && viewport->is_inside_tree()) {
+
+	if (!viewport || !viewport->is_inside_tree()) {
+		return;
+	}
+
+	if (!custom_viewport && !ObjectDB::get_instance(custom_viewport_id)) {
 		viewport->assign_next_enabled_camera_2d(group_name);
 	}
 }
 
 bool Camera2D::is_current() const {
-	if (viewport && !(custom_viewport && !ObjectDB::get_instance(custom_viewport_id))) {
+	if (!viewport) {
+		return false;
+	}
+
+	if (!custom_viewport && !ObjectDB::get_instance(custom_viewport_id)) {
```

Comment:
Is this some refactoring? It changes logic and looks wrong. If anything, it should be
```suggestion
	if (!custom_viewport || ObjectDB::get_instance(custom_viewport_id)) {
```